keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/KDEVertex.java

```diff
@@ -0,0 +1,135 @@
+package io.improbable.keanu.vertices.dbl.probabilistic;
+
+
+import io.improbable.keanu.distributions.continuous.Gaussian;
+import io.improbable.keanu.distributions.continuous.Uniform;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class KDEVertex extends ProbabilisticDouble {
+    public double bandwidth;
+    private DoubleTensor samples;
+
+    public KDEVertex(DoubleTensor samples, double bandwidth){
+        this.samples = samples;
+        this.bandwidth = bandwidth;
+    }
+
+    public KDEVertex(DoubleTensor samples) {
+        this.samples = samples;
+        this.bandwidth = scottsBandwith();
```

Comment:
`this(samples, scottsBandwith())` is preferable, to avoid code duplication and potential for bugs if the code changes later

Reply:
good point, I'll have to make scottsBandwidth() a static method